File: redome/src/main/java/br/org/cancer/modred/controller/dto/PacienteAvaliacaoTarefaDTO.java
package br.org.cancer.modred.controller.dto;

import com.fasterxml.jackson.annotation.JsonView;

import br.org.cancer.modred.controller.view.AvaliacaoView;
import br.org.cancer.modred.model.Cid;

/**
 * Classe DTO que representa um item na tela de pacientes em avaliação.
 * 
 * @author Fillipe Queiroz
 *
 */
public class PacienteAvaliacaoTarefaDTO {

	@JsonView(AvaliacaoView.ListarAvaliacoesPacientes.class)
	private Long rmr;
	@JsonView(AvaliacaoView.ListarAvaliacoesPacientes.class)
	private String nome;
	@JsonView(AvaliacaoView.ListarAvaliacoesPacientes.class)
	private Integer idade;
	@JsonView(AvaliacaoView.ListarAvaliacoesPacientes.class)
	private String tempoCadastro;
	@JsonView(AvaliacaoView.ListarAvaliacoesPacientes.class)
	private Cid cid;

	public PacienteAvaliacaoTarefaDTO() {}

	/**
	 * Construtor com parametros rmr, nome e tempoCadastro.
	 * 
	 * @param rmr - rmr do paciente
	 * @param nome - nome do paciente
	 * @param tempoCadastro - tempo desde que paciente foi cadastrado
	 */
	public PacienteAvaliacaoTarefaDTO(Long rmr, String nome, String tempoCadastro) {
		this.rmr = rmr;
		this.nome = nome;
		this.tempoCadastro = tempoCadastro;
	}

	/**
	 * RMR do paciente associado a classe.
	 * 
	 * @return RMR do paciente.
	 */
	public Long getRmr() {
		return rmr;
	}

	/**
	 * Seta o rmr do paciente.
	 * 
	 * @param rmr
	 */
	public void setRmr(Long rmr) {
		this.rmr = rmr;
	}

	/**
	 * Retorna o nome do paciente associado a classe.
	 * 
	 * @return nome do paciente.
	 */
	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	/**
	 * Seta a idade do paciente.
	 * 
	 * @return
	 */
	public Integer getIdade() {
		return idade;
	}

	/**
	 * Seta a idade do paciente.
	 * 
	 * @param idade
	 */
	public void setIdade(Integer idade) {
		this.idade = idade;
	}

	/**
	 * Retorna o aging da data de cadastro.
	 * 
	 * @return tempoCadastro
	 */
	public String getTempoCadastro() {
		return tempoCadastro;
	}

	/**
	 * Seta o aging da data de cadastro do paciente.
	 * 
	 * @param tempoCadastro
	 */
	public void setTempoCadastro(String tempoCadastro) {
		this.tempoCadastro = tempoCadastro;
	}

	/**
	 * Retorna a cid.
	 * 
	 * @return
	 */
	public Cid getCid() {
		return cid;
	}

	/**
	 * Seta uma cid.
	 * 
	 * @param cid
	 */
	public void setCid(Cid cid) {
		this.cid = cid;
	}

}

File: redome-front/src/app/paciente/consulta/avaliacao/paciente.avaliacao.tarefa.dto.ts
import { Cid } from '../../../shared/dominio/cid';
/**
 * Classe que representa um PacienteAvaliacaoTarefaDTO dto
 * 
 * @author Fillipe Queiroz
 * @export
 */
export class PacienteAvaliacaoTarefaDTO {

	private _rmr: number;
	private _nome:string;
	private _idade: number;
	private _tempoCadastro:string;
	private _cid:Cid;
	private _descricaoCid:string;

	/**
	 * Retorna o rmr
	 */
	public get rmr(): number {
		return this._rmr;
	}

	/**
	 * Seta o rmr
	 */
	public set rmr(value: number) {
		this._rmr = value;
	}

	/**
	 * Recupera o nome
	 */
	public get nome(): string {
		return this._nome;
	}

	/**
	 * Seta o nome
	 */
	public set nome(value: string) {
		this._nome = value;
	}

	/**
	 * Recupera a idade
	 */
	public get idade(): number {
		return this._idade;
	}

	/**
	 * Seta a idade
	 */
	public set idade(value: number) {
		this._idade = value;
	}

	/**
	 * Recupera o tempo de cadastro
	 */
	public get tempoCadastro(): string {
		return this._tempoCadastro;
	}

	/**
	 * Seta o tempo de cadastro
	 */
	public set tempoCadastro(value: string) {
		this._tempoCadastro = value;
	}

	/**
	 * Recupera a cid.
	 */
	public get cid(): Cid {
		return this._cid;
	}

	/**
	 * Seta a cid.
	 */
	public set cid(value: Cid) {
		this._cid = value;
	}


}
